packages/static-build/test/builds.test.js

```diff
@@ -290,6 +290,54 @@ it(
   FOUR_MINUTES
 );
 
+it(
+  'Should build Gatsby with configuration defined in esm',
+  async () => {
+    const { workPath } = await runBuildLambda(
+      path.join(__dirname, 'build-fixtures/14-gatsby-with-esm-config')
+    );
+
+    const contents = await fs.readdir(workPath);
+
+    expect(contents.some(name => name === 'gatsby-config.js')).toBeFalsy();
+    expect(contents.some(name => name === 'gatsby-config.ts')).toBeFalsy();
+    expect(contents.some(name => name === 'gatsby-config.mjs')).toBeTruthy();
+    // using `import` causes a seg fault.
+    expect(await fs.readFile(path.join(workPath, 'gatsby-config.mjs'), 'utf-8'))
+      .toBe(`import userConfig from "./gatsby-config.mjs.__vercel_builder_backup__.mjs";
```

Comment:
**suggestion (non-blocking):** make assertion less specific

This test assertion is brittle when we want to change the injected code. What do you think about asserting only the first line, which captures the core of the difference between this type of injection vs. the others?